### DISCUSSION TITLE
Username / email

### DISCUSSION THREAD

**Discussion by gardiol** [immich-app/immich#6491] [General]
Hi!
why does immich require email as username for login?

in my system i have a few users (user1, user2, user3...) which authenticate with a simple "username" (not an email address) and password. For Immich, i need to have the same username, not a different one (with a fake or real email address).

Would this "email" address as usename be avoided by using OAuth? 
Can it be disabled also for normal logins?

Is it possible, in short, to have simple usernames and not emails for login?

**Comment by tubbadu** [comment]
I agree, there is no need to force the use of an email.

**Comment by K4LCIFER** [comment]
It feels especially weird for the admin account to require an email. You end up in a situation where you create the admin account with your email, but then you cannot use that email for your personal non-admin account.

**Comment by TheGorf** [comment]
Came here to ask this exact same question. My instance will never be open to anyone beyond my wife and I. I have zero need for email based usernames. I really just want simple usernames as an option. Email is clunky. Additionally, from a cyber-security perspective it is generally not considered best practice to use PII data as a the identifying component of a user based system.

**Comment by Gitonthegrid** [comment]
Same here. There is really no need to ask for this. I really don't like accounts with email, and especially self hosted stuff

**Comment by samholton** [comment]
The email validation rules allow email without top level domain. So you could do something like user@local or user@family or whatever as a work around.

**Reply by K4LCIFER** [reply]
The fact that that is an option makes the entire situation all the more nonsensical to me.

**Reply by mertalev** [reply]
This was added via [this](https://github.com/immich-app/immich/pull/2762) PR, since it's possible to have emails without a TLD or where the domain is also the TLD.

**Reply by gardiol** [reply]
I see but it's only a wachy hack since my users are all the form "wife" "son1" and so on, adding "@" still make no sense and will be even more awkward since wife@local impacts the WSF even worse than using her real email.

**Comment by gardiol** [comment]
To be honest, why don't any of us open a pull request and fix it?

**Reply by TheGorf** [reply]
Because I know nothing of TypeScript and I'm still trying to make heads or tails of the code.

**Comment by mxz94** [comment]
I agree, there is no need to force the use of an email.

**Comment by gardiol** [comment]
Well, it looks like the devs want to use an email, they even went to length of actually testing that an email is provided. So i guess opening a PR would be pointless unless some of the dev team acknowledge this discussion and agrees with all of us.

**Comment by schuhbacca** [comment]
My guess is that it makes oauth linkage easier.

**Reply by schuhbacca** [reply]
Actually found this reply in discord: 
![image](https://github.com/immich-app/immich/assets/5739463/4bba4d48-1283-4316-9b40-ad66a3da60af)

**Reply by gardiol** [reply]
I hope they will give users choices one day.

**Comment by naworm** [comment]
the worse part is that we can't edit the email afterward, wtf ?

**Reply by alextran1502** [reply]
You can edit the email from the User Settings page from the web

**Reply by naworm** [reply]
I just saw that. I think there was a situation where I was unable to edit it. My bad.

**Comment by PANICBUTT0N** [comment]
Seconding this. It feels weird to me to require an email for a self-hosted service. I would appreciate it if accounts could be created with usernames and an optional email address if you would like to enable email notifications.

**Comment by steveskarn** [comment]
I agree

**Comment by RealSkydiver** [comment]
Same here — and in my case it’s even worse. I’ve tested Immich quite extensively (uploaded the first 30k photos out of roughly 250k), and overall it’s not bad so far. However, my wife really dislikes the login process, and I couldn’t convince her otherwise. In the end, she simply won’t use it because of that.

It’s also worth mentioning that both of us want to switch between our accounts (we don’t have any secrets from each other). That makes this kind of login mechanism even more of a hassle. A simple “first name / password” approach would solve the problem entirely.

I’m a software architect (mostly in embedded systems), and from my perspective there’s no real reason — apart from personal developer preferences — to enforce login via email address. Especially in a local setup, this kind of security pattern becomes largely unnecessary.

On top of that, there’s no way to configure password complexity, but a fixed minimum length of 8 characters is enforced. That’s an arbitrary constraint — especially for a self-hosted setup.

**Comment by flemmingss** [comment]
I Agree on this.
I have maybe 50 users on my server. As a  workaround I just call all users username@no.mail 
But if I could skip the @no.mail ending it sure will be a lot easyer for my users.
